src/core/lib/security/context/security_context.cc

```diff
@@ -56,12 +56,12 @@ grpc_call_error grpc_call_set_credentials(grpc_call* call,
     LOG(ERROR) << "Method is client-side only.";
     return GRPC_CALL_ERROR_NOT_ON_SERVER;
   }
-  ctx = static_cast<grpc_client_security_context*>(
-      grpc_call_context_get(call, GRPC_CONTEXT_SECURITY));
+  auto* arena = grpc_call_get_arena(call);
+  ctx = grpc_core::DownCast<grpc_client_security_context*>(
```

Comment:
Can't this just be `arena->GetContext<grpc_client_security_context>()`, due to the `ContextSubclass` template overrides in the .h file?

Reply:
Mentioned elsewhere... the subclassing stuff only works for the free function GetContext... the arena functions are the building blocks for that - ultimately we don't expect much usage, but whilst we have filters we kind of do